applications/OptimizationApplication/python_scripts/controls/material/material_properties_control.py

```diff
@@ -0,0 +1,102 @@
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+from KratosMultiphysics.OptimizationApplication.controls.control import Control
+from KratosMultiphysics.OptimizationApplication.utilities.optimization_problem import OptimizationProblem
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import ContainerExpressionTypes
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import SupportedSensitivityFieldVariableTypes
+from KratosMultiphysics.OptimizationApplication.utilities.helper_utilities import IsSameContainerExpression
+
+def Factory(model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+    return MaterialPropertiesControl(model, parameters, optimization_problem)
+
+class MaterialPropertiesControl(Control):
+    """Material properties control
+
+    This is a generic material properties control which creates a control
+    for the specified control variable. This does not do any filtering.
+
+    TODO: Extend with filtering techniques when they are implemented.
+
+    """
+    def __init__(self, model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+        super().__init__()
+
+        default_settings = Kratos.Parameters("""{
+            "combined_output_model_part_name": "<CONTROL_NAME>_combined_no_neighbours_control",
+            "model_part_names"               : [""],
+            "control_variable_name"          : ""
```

Comment:
control_variable_name should be specified by the control and it should not be an input. So it should not be given as input because the control implementation and type decides about it. since we have list of model parts let's call the input "model_parts_names"

Reply:
I would not make room for "combined_output_model_part_name". The control model part and its output should be implemented explicitly by control.

Reply:
The control variable name is a variable for `SystemIndetification` hence, i would keep it for this control. For specific controls, we can remove this `control_variable_name`. This is not for all the controls.

I need the output `combined_output_model_part_name`  which to create the intersected and merged model parts so if the user wants, then they can re-use model parts. (and outputing to VTK will be really nice with nice names).

The defaults are there in most cases, and user won't have to touch the defaults.